coffee/base.py

```diff
@@ -151,7 +161,7 @@ def __deepcopy__(self, memo):
         return self.__class__(dcopy(self.children[0]), dcopy(self.children[1]))
 
     def gencode(self, not_scope=True, parent=None):
-        subtree = (" "+self.op+" ").join([n.gencode(not_scope, self) for n in self.children])
+        subtree = (" "+type(self).op+" ").join([n.gencode(not_scope) for n in self.children])
```

Comment:
I assume you use `type(self).op` because `self.op` did not work, yet on the toy examples I have tried, `self.op` works just fine. I'm interested in what cases `self.op` fails.